t/t9701-perl-git-MSWin32.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+#
+# Copyright (c) 2018 Chris Lindee
+#
+
+test_description='perl interface (Git.pm) on MSWin32'
+. ./test-lib.sh
+
+find_MSWin32_perl() {
+	local perl
+	for perl in $(which -a perl); do
+		if "$perl" -e 'exit 1 if $^O ne q{MSWin32}'; then
+			echo "$perl"
+			return 0
+		fi
+	done
+	return 1
+}
```

Comment:
To make sure that I describe my idea thoroughly: I am talking about something like this:

```sh
# Some tests require a pure Win32 Perl interpreter, such as
# Strawberry Perl or ActiveState Perl. To run these tests, the
# `GIT_MSWIN32_PERL_PATH` variable must be set to the
# directory containing the `perl.exe`.
test_lazy_prereq HAVE_MSWIN32_PERL '
    test -x "$GIT_MSWIN32_PERL_PATH/perl.exe" &&
    perl -e 'exit $^O -eq "MSWin32";'
'
```

The test cases would then use that prerequisite and prefix `PATH`:

```sh
test_expect_success HAVE_MSWIN32_PERL 'test quoting with MSWin32 Perl' '
    PATH="$GIT_MSWIN32_PERL_PATH:$PATH" &&
    export PATH &&
    ...
'
```

Reply:
How does `test -x` evaluate whether a file is executable on Windows?  It looks like it checks for a specific extension on the file.  It seems like anything matching `*.exe` is flagged as executable.

Reply:
It first checks for the `.exe` extension, then opens the file to look whether the file starts with a `#!<...>` line.